lib/WebLedgerClient.js

```diff
@@ -0,0 +1,288 @@
+/*!
+ * Copyright (c) 2018-2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const constants = require('./constants');
+const {create} = require('apisauce');
+const https = require('https');
+const WebLedgerClientError = require('./WebLedgerClientError');
+
+class WebLedgerClient {
+  /**
+   * @param {string} hostname - the hostname of the ledger node.
+   * @param {boolean} [strictSSL=true] - validate the host's SSL certificate.
+   * @param [logger] - the logger to use.
+   * @param {Agent} [httpsAgent] - a custom `https.Agent` instance.
+   */
+  constructor({httpsAgent, hostname, logger, strictSSL = true}) {
+    this.hostname = hostname;
+    if(!hostname) {
+      throw new TypeError('Missing ledger hostname.');
+    }
+
+    this.logger = logger || console;
+
+    this.httpsAgent = httpsAgent;
+    if(!(strictSSL || this.httpsAgent)) {
+      this.httpsAgent = new https.Agent({rejectUnauthorized: false});
+    }
+    this.ledgerAgent = null;
+    this.ledgerAgentStatus = null;
+  }
+
+  /**
+   * Fetches a Record for a given ID.
+   *
+   * @param {string} id - the ID URI for the record.
+   *
+   * @returns {Promise<object>} - the record object.
+   */
+  async getRecord({id}) {
+    if(!id) {
+      throw new TypeError('Invalid or missing Record URI.');
+    }
+
+    const ledgerAgent = await this.getAgent();
+    const baseURL = ledgerAgent.service.ledgerQueryService;
+    const queryServiceApi = create({baseURL});
+
+    this.logger.log(`Retrieving Record [ledger: ${this.hostname}]:`, id);
+
+    const response = await queryServiceApi.post(
+      `?id=${encodeURIComponent(id)}`, {}, {
+        httpsAgent: this.httpsAgent,
+        headers: constants.DEFAULT_HEADERS
+      });
+
+    if(response.problem) {
+      if(response.status === 404) {
+        throw new WebLedgerClientError(
+          'Record not found.', 'NotFoundError',
+          {baseURL, id, status: response.status});
+      }
+      const error = new WebLedgerClientError(
+        'Error retrieving record.', 'NetworkError');
+      if(response.problem === 'CLIENT_ERROR') {
+        error.details = {
+          baseURL, id, error: response.data, status: response.status
+        };
+      } else {
+        error.details = {
+          baseURL, id, error: response.originalError, status: response.status
+        };
+      }
+      throw error;
+    }
+    const body = response.data;
+
+    const result = {
+      found: true,
+      retry: false,
+      id,
+      hostname: this.hostname,
+      meta: body.meta
+    };
+
+    const {record} = body;
+
+    if(!record) {
+      throw new WebLedgerClientError(
+        'Fetched Record has empty body/"record" property.', 'DataError',
+        {baseURL, id});
+    }
+
+    // full Record
+    result.type = 'WebLedgerRecord'; //FIXME: check if type is correct
```

Comment:
If we don't have a use for this yet, let's remove it. I find this type name to be confusing anyway -- as we're using a property "record" to point at the actual record here.

Reply:
Removed, Done.